log/mcap.ksy

```diff
@@ -0,0 +1,397 @@
+meta:
+  title: MCAP
+  id: mcap
+  file-extension: mcap
+  license: Apache-2.0
+  endian: le
+doc: |
+  MCAP is a modular container format and logging library for pub/sub messages with
+  arbitrary message serialization. It is primarily intended for use in robotics
+  applications, and works well under various workloads, resource constraints, and
+  durability requirements.
+doc-ref: https://github.com/foxglove/mcap
+seq:
+  - id: header_magic
+    type: magic
+
+  - id: records
+    type: record
+    repeat: until
+    repeat-until: _.op == opcode::footer
+
+  - id: footer_magic
+    type: magic
+
+instances:
+  footer:
+    pos: "_io.size - sizeof<magic> - sizeof<footer> - sizeof<record_prefix>"
+    size-eos: true
+    type: record
+
+enums:
+  opcode:
+    0x01: header
+    0x02: footer
+    0x03: schema
+    0x04: channel
+    0x05: message
+    0x06: chunk
+    0x07: message_index
+    0x08: chunk_index
+    0x09: attachment
+    0x0a: attachment_index
+    0x0b: statistics
+    0x0c: metadata
+    0x0d: metadata_index
+    0x0e: summary_offset
+    0x0f: data_end
+
+types:
+  magic:
+    seq:
+      - id: magic
+        contents: [0x89, "MCAP0\r\n"]
+
+  prefixed_str:
+    seq:
+      - id: len
+        type: u4
+      - id: str
+        type: str
+        size: len
+        encoding: UTF-8
+
+  tuple_str_str:
+    seq:
+      - id: key
+        type: prefixed_str
+      - id: value
+        type: prefixed_str
+
+  map_str_str:
+    types:
+      entries:
+        seq:
+          - id: entry
+            type: tuple_str_str
+            repeat: eos
+    seq:
+      - id: len
+        type: u4
+      - id: entry
+        size: len
+        type: entries
+
+  records:
+    seq:
+      - id: records
+        type: record
+        repeat: eos
+
+  record_prefix:
```

Comment:
This is a unused type only referenced in `sizeof<>`, which is disturbing. And when I look at individual pos calculations which involve `sizeof<record_prefix>`, I doubt that they're really necessary.

Reply:
Does this mean you would recommend replacing them with hard-coded numbers like `8 + 1`? Or is there some way to reference the actually used fields, e.g. `sizeof<record.op> + sizeof<record.len>`? I tried this (and `::`) but it didn't seem to work.